src/aws/osml/model_runner/app.py

```diff
@@ -786,7 +845,56 @@ def select_features(
             feature_distillation_option_dict = json.loads(image_request_item.feature_distillation_option)
             feature_distillation_option = FeatureDistillationDeserializer().deserialize(feature_distillation_option_dict)
             feature_selector = FeatureSelector(feature_distillation_option)
-            return feature_selector.select_features(features)
+
+            region_size = ast.literal_eval(ServiceConfig.region_size)
+            tile_size = ast.literal_eval(image_request_item.tile_size)
+            overlap = ast.literal_eval(image_request_item.tile_overlap)
+
+            # This code applies a feature selector only to the features that came from regions of the image
+            # that were processed multiple times. First features are grouped based on the region they were
+            # processed in. Any features found in the overlap area between regions are run through the
+            # FeatureSelector. If they were not part of an overlap area between regions, they will be grouped
+            # based on tile boundaries. Any features that fall into the overlap of adjacent tiles are filtered
+            # by the FeatureSelector. All other features should not be duplicates; they are added to the result
+            # without additional filtering.
+            #
+            # Computationally this implements two critical factors that lower the overall processing time for the
+            # O(N^2) selection algorithms. First it will filter out the majority of features that couldn't possibly
+            # have duplicates generated by our tiled image processing, Second it runs the selection algorithms
+            # incrementally on much smaller groups of features.
```

Comment:
This should probably be a pydoc comment at the beginning of the function instead of a huge comment block in the middle.

Reply:
Will fix